source/dofs/dof_handler.cc

```diff
@@ -2431,10 +2438,10 @@ DoFHandler<dim, spacedim>::set_fe(const hp::FECollection<dim, spacedim> &ff)
         {
           hp_capability_enabled = false;
 
-          // unsubscribe connections to signals
-          for (auto &connection : this->tria_listeners)
+          // unsubscribe connections to signals that are relevant for hp-mode
+          for (auto &connection : this->tria_listeners_for_transfer)
```

Comment:
It took me a little bit to see why this is possible -- because you aren't actually in hp mode any more. Maybe say that in the comment: "...that are only relevant for hp-mode, but we only have a single element here" (emphasis on *only*)